### PR TITLE
Add binding suggestion invocation to VsHost after solution opens

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Implements #4969

### CODE REVIEW

File: src/Integration/MefServices/VsSessionHost.cs [hunk 1]
```diff
@@ -84,8 +95,11 @@ internal sealed class VsSessionHost : IHost, IProgressStepRunnerWrapper, IDispos
             this.configurationProvider = configurationProvider ?? throw new ArgumentNullException(nameof(configurationProvider));
             this.sharedBindingConfigProvider = sharedBindingConfigProvider;
             this.credentialStoreService = credentialStoreService;
+            this.connectedModeWindowEventListener = connectedModeWindowEventListener;
+            this.sharedBindingSuggestionService = sharedBindingSuggestionService;
             this.solutionTracker.ActiveSolutionChanged += this.OnActiveSolutionChanged;
             this.Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            connectedModeWindowEventListener.SubscribeToConnectedModeWindowEvents(this);
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:102]: Event subscription for executing autobind after TE loads

File: src/Integration/MefServices/VsSessionHost.cs [hunk 2]
```diff
@@ -208,9 +222,11 @@ private void ResetBinding(bool abortCurrentlyRunningWorklows, bool clearCurrentB
             if (clearCurrentBinding || bindingConfig == null || bindingConfig.Mode == SonarLintMode.Standalone)
             {
                 this.ClearCurrentBinding();
-                SharedBindingConfig = sharedBindingConfigProvider.GetSharedBinding();
-                VisualStateManager.HasSharedBinding = SharedBindingConfig != null;
-                VisualStateManager.ResetConnectionConfiguration();
+
+                if (!clearCurrentBinding) // when solution is open, but unbound
+                {
+                    UpdateSharedBindingSuggestion();
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:228]: Move shared binding update to a separate method

File: src/Integration/MefServices/VsSessionHost.cs [hunk 3]
```diff
@@ -208,9 +222,11 @@ private void ResetBinding(bool abortCurrentlyRunningWorklows, bool clearCurrentB
             if (clearCurrentBinding || bindingConfig == null || bindingConfig.Mode == SonarLintMode.Standalone)
             {
                 this.ClearCurrentBinding();
-                SharedBindingConfig = sharedBindingConfigProvider.GetSharedBinding();
-                VisualStateManager.HasSharedBinding = SharedBindingConfig != null;
-                VisualStateManager.ResetConnectionConfiguration();
+
+                if (!clearCurrentBinding) // when solution is open, but unbound
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:226]: We don't want to show the goldbar when there's no solution open

File: src/Integration/MefServices/VsSessionHost.cs [hunk 4]
```diff
@@ -228,6 +244,16 @@ private void ResetBinding(bool abortCurrentlyRunningWorklows, bool clearCurrentB
...
         }
 
+        private void UpdateSharedBindingSuggestion()
+        {
+            SharedBindingConfig = sharedBindingConfigProvider.GetSharedBinding();
+
+            VisualStateManager.HasSharedBinding = SharedBindingConfig != null;
+            VisualStateManager.ResetConnectionConfiguration();
+
+            sharedBindingSuggestionService.Suggest(SharedBindingConfig.GetServerType(),() => this.ActiveSection?.ConnectCommand);
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:254]: ConnectCommand will be replaced by non-null value after initialization. In order for delayed execution to pick that up, I used a lambda

File: src/Integration/MefServices/VsSessionHost.cs [hunk 5]
```diff
@@ -285,6 +311,7 @@ private void Dispose(bool disposing)
                 if (disposing)
                 {
                     this.solutionTracker.ActiveSolutionChanged -= this.OnActiveSolutionChanged;
+                    connectedModeWindowEventListener.Dispose();
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration/MefServices/VsSessionHost.cs:314]: Unsubscription from host events

**Review by ugras-ergun-sonarsource — APPROVED** [review]

**Pull request merged**